### DISCUSSION TITLE
Typing issue: Words registered in the user dictionary do not appear as conversion candidates for the second or subsequent chunks.

### DISCUSSION THREAD

**Discussion by mikunimaru** [google/mozc#807] [Q&A]
## Category of the typing issue
Choose one of them (delete rest of them)
3. Conversion issue (e.g. "夕日がきれい" is not converted as expected).

## Issues
For example, I assume the word "〒" has been registered in the user dictionary as "てすと."

| input [e.g.ゆうひ] | expected [e.g. 夕日] | actual [e.g. ユウヒ] |
| ------------ | ------------ | ------------|
| てすとてすと | 〒〒 | 〒テスト |

## Version or commit-id
Mozc-2.29.5210.100+24.11.oss

**Comment by yukawa** [comment]
> For example, I assume the word "〒" has been registered in the user dictionary as "てすと."
> | input [e.g.ゆうひ] | expected [e.g. 夕日] | actual [e.g. ユウヒ] |
> | ------------ | ------------ | ------------|
> | てすとてすと | 〒〒 | 〒テスト |

I could not reproduce the issue in f79e092ac7068f6bbdacb2b069641771e7cb1f43.

### Steps to reproduce
1. Install Mozc
2. Register "〒" as reading: "てすと", part-of-speech: Noun
3. Open Notebook
4. Type "てすとてすと"
5. Hit space key
6. Hit space key again

### Actual result
* At the step 4, the suggestion candidate is "テストテスト"
<img width="85" alt="image" src="https://github.com/google/mozc/assets/5861397/e903e1c4-4c2e-47f4-b3fa-8eb3f306bf94">

* At the step 5, the conversion result becomes "テストテスト"
<img width="63" alt="image" src="https://github.com/google/mozc/assets/5861397/9786b9b9-54a1-4c0b-99b8-ed57231f62e0">

* At the step 6, the conversion result becomes "〒テスト"
<img width="146" alt="image" src="https://github.com/google/mozc/assets/5861397/b959eab3-6c97-43ff-a625-3bb5156c71bd">

I suspect that what you saw was affected by your previous conversion. To confirm, you can open the Settings dialog and click "Clear personalization data". (caution: you would lose your typing history by doing this)

**Comment by mikunimaru** [comment]
I apologize for any confusion due to insufficient explanation. 
To provide better clarity, I would like to present the issue once again, this time using images.

![1](https://github.com/google/mozc/assets/43168745/ee43dfb7-6c0b-4c80-9056-0fb3d9b9671b)
![2](https://github.com/google/mozc/assets/43168745/60d99f1f-3b29-4b28-88e4-1981941612ad)

In this way, words registered in the dictionary will only be displayed in the first chunk.
It is impossible to output "〒〒".

**Reply by yukawa** [reply]
Hmm, it's working well in my environment.
<img width="146" alt="image" src="https://github.com/google/mozc/assets/5861397/73359017-ad5e-4765-95d4-640d18b02bfc">

Can you update to f79e092ac7068f6bbdacb2b069641771e7cb1f43 then clear typing history to see if the issue persists?

**Comment by mikunimaru** [comment]
After updating the version and checking again, I found the cause. It seems to occur when the part of speech of the registered word is "記号," "独立語," or "顔文字".
![1](https://github.com/google/mozc/assets/43168745/700d6af7-abd2-42ca-b9ac-a7df7e75713d)
![2](https://github.com/google/mozc/assets/43168745/b88d4614-c138-4432-b4be-05716334ef47)
![3](https://github.com/google/mozc/assets/43168745/01e0383c-b2f5-44bb-bc1f-7b5255d47ce1)

I apologize for the lack of information in my report as a result of trying to generalize.

I originally discovered this issue when I registered "®" as a symbol and could not convert it to "xxxx®".
Since the normal conversion allows symbols to be converted anywhere in a sentence, so at least the behavior of symbol conversion seems a bit inconsistent. This is inconvenient when registering currency symbols, etc.

**Reply by yukawa** [reply]
Thank you for figuring out the condition to reproduce the issue.

> Since the normal conversion allows symbols to be converted anywhere in a sentence, so at least the behavior of symbol conversion seems a bit inconsistent. This is inconvenient when registering currency symbols, etc.

Right. For your case, instead of adding a new symbol and/or symbol reading to user dictionary, adding them to TSV files under [src/data/symbol/](https://github.com/google/mozc/tree/master/src/data/symbol) then rebuild and install Mozc should give the consistent result as other symbol conversions.

For example of ®:
https://github.com/google/mozc/blob/f79e092ac7068f6bbdacb2b069641771e7cb1f43/src/data/symbol/symbol.tsv#L547

For example of currency symbols:
https://github.com/google/mozc/blob/f79e092ac7068f6bbdacb2b069641771e7cb1f43/src/data/symbol/symbol.tsv#L789-L821

I agree that it'd be nice if user dictionary entries whose part-of-speech are symbol-like can be handled as if they were already included in the above TSV files. I can't promise whether we actually implement such a behavior in the near future, but I'll definitely bring this to the team to discuss further.

Thanks!
